lib/tools/adb-commands.js

```diff
@@ -835,10 +835,17 @@ methods.push = async function push (localPath, remotePath, opts) {
  *
  * @param {string} remotePath - The source path on the remote device.
  * @param {string} localPath - The destination path to the file on the local file system.
+ * @param {object} opts - Additional options mapping. See
+ *                        https://github.com/appium/node-teen_process,
+ *                        _exec_ method options, for more information about available
+ *                        options.
  */
-methods.pull = async function pull (remotePath, localPath) {
+methods.pull = async function pull (remotePath, localPath, opts) {
   // pull folder can take more time, increasing time out to 60 secs
-  await this.adbExec(['pull', remotePath, localPath], {timeout: 60000});
+  if (opts.timeout === null) {
```

Comment:
Please do not mutate opts. Consider using opts as following to avoid mutation:

```
await this.adbExec(['pull', remotePath, localPath], {
  ...opts,
  timeout: opts.timeout ?? 60000,
})
```

Reply:
Fixed 👍